client/my-sites/stats/features/modules/stats-locations/stats-locations.tsx

```diff
@@ -102,15 +105,25 @@ const StatsLocations: React.FC< StatsModuleLocationsProps > = ( { query, summary
 	const geoMode = GEO_MODES[ selectedOption ];
 	const title = optionLabels[ selectedOption ]?.selectLabel;
 
+	const { supportsLocationsStats: supportsLocationsStatsFeature } = useSelector( ( state ) =>
+		getEnvStatsFeatureSupportChecks( state, siteId )
+	);
+
 	// Main location data query
 	const {
-		data = [],
+		data: locationsViewsData = [],
 		isLoading: isRequestingData,
 		isError,
 	} = useLocationViewsQuery< StatsLocationViewsData >( siteId, geoMode, query, countryFilter, {
-		enabled: ! shouldGate,
+		enabled: ! shouldGate && supportsLocationsStatsFeature,
 	} );
 
+	const countriesViewsData = useSelector( ( state ) =>
+		getSiteStatsNormalizedData( state, siteId, statType, query )
+	) as [ id: number, label: string ];
+
+	const data = supportsLocationsStatsFeature ? locationsViewsData : countriesViewsData;
+
 	// Only fetch separate countries list if we're not already in country tab
 	// This is to avoid fetching the same data twice.
 	const { data: countriesList = [] } = useLocationViewsQuery< StatsLocationViewsData >(
```

Comment:
While testing, I realized that we are still trying to fetch the new locations API when switching to regions or cities. I think you should also add `supportsLocationsStatsFeature` to the enabled prop of this `useLocationViewsQuery` call. It is intended to populate the countries filter.

![CleanShot 2025-02-04 at 18 29 08@2x](https://github.com/user-attachments/assets/87455020-80de-43fc-88df-9b96f5925239)

Reply:
Nice catch, I had missed this second usage of `useLocationViewsQuery`.